src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java

```diff
@@ -223,35 +223,42 @@ public boolean canBeCreatedBy(Auth0UserProfile profile) {
             userId = otpUser.id;
         } else {
             // If userId was provided, follow authorization provided by canBeManagedBy
-            return canBeManagedBy(profile);
+            return canBeManagedBy(requestingUser);
         }
-        return super.canBeCreatedBy(profile);
+        return super.canBeCreatedBy(requestingUser);
     }
 
     /**
      * Confirm that the requesting user has the required permissions
      */
     @Override
-    public boolean canBeManagedBy(Auth0UserProfile user) {
+    public boolean canBeManagedBy(RequestingUser requestingUser) {
         // This should not be possible, but return false on a null userId just in case.
         if (userId == null) return false;
-        // If the user is attempting to update someone else's monitored trip, they must be admin.
+        // If the user is attempting to update someone else's monitored trip, they must be admin or an API user if the
+        // OTP user is assigned to that API.
         boolean belongsToUser = false;
         // Monitored trip can only be owned by an OtpUser (not an ApiUser or AdminUser).
-        if (user.otpUser != null) {
-            belongsToUser = userId.equals(user.otpUser.id);
+        if (requestingUser.otpUser != null) {
+            belongsToUser = userId.equals(requestingUser.otpUser.id);
         }
 
         if (belongsToUser) {
             return true;
-        } else if (user.adminUser != null) {
+        } else if (requestingUser.isThirdPartyUser()) {
+            // get the required OTP user to confirm they are associated with the requesting API user.
+            OtpUser otpUser = Persistence.otpUsers.getById(userId);
+            if (otpUser != null && requestingUser.apiUser.id.equals(otpUser.applicationId)) {
+                return true;
+            }
+        } else if (requestingUser.adminUser != null) {
```

Comment:
This can become `requestingUser.isAdminI()`.

Reply:
Updated.